py_backwards/transformers/async_await.py

```diff
@@ -0,0 +1,64 @@
+from typed_ast import ast3 as ast
+from .base import BaseNodeTransformer
+
+
+ASYNCIO_MODULE = '__py_backwards_asyncio__'
+
+def asyncio_decorator():
+    return ast.Attribute(
+        value=ast.Name(
+            id=ASYNCIO_MODULE,
+            ctx=ast.Load(),
+        ),
+        attr='coroutine',
+    )
+
+
+def splice(fn, vars):
+    for i, v in enumerate(vars):
+        nv = fn(v)
+        if nv is None:
+            yield v
+        else:
+            yield nv
+            for v in vars[i:]:
+                yield v
+            break
+
+
+class AsyncAwaitTransformer(BaseNodeTransformer):
+    """Compiles:
+        async def ham():
+            await foo()
+    To
+
+        import asyncio as __py_backwards_asyncio__
+
+        @__py_backwards_asyncio__.coroutine
+        def ham():
+            yield from foo()
+    """
+    target = (3, 4)
+
+    def visit_Module(self, n: ast.Module) -> ast.Module:
+        self._tree_changed = True
+        def dosplice(v):
+            # insert just after the first node that's not a future import.
+            if not (isinstance(v, ast.ImportFrom) and v.module == '__future__'):
+                return ast.Import(names=[ast.alias('asyncio', ASYNCIO_MODULE)])
+
+        return self.generic_visit(ast.Module(body=list(splice(dosplice, n.body))))
+
+    def visit_AsyncFunctionDef(self, node: ast.AsyncFunctionDef) -> ast.FunctionDef:
```

Comment:
Looks like a bug in typed ast.
Maybe it would be better to use lower-level api, like in https://github.com/nvbn/py-backwards/blob/master/py_backwards/transformers/variables_annotations.py#L20